source/common/buffer/watermark_buffer.cc

```diff
@@ -195,10 +215,11 @@ BufferMemoryAccountImpl::createAccount(WatermarkBufferFactory* factory,
 }
 
 int BufferMemoryAccountImpl::balanceToClassIndex() {
-  const uint64_t shifted_balance = buffer_memory_allocated_ >> 20; // shift by 1MB.
+  static uint32_t bitshift = factory_->bitshift();
```

Comment:
Do you mean const?
Use of static here is sure to cause problems in tests as test cases beyond the first would effectively ignore config changes.